File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.java
class Solution {
    public int findCenter(int[][] edges) {
        int n=edges.length;
    int degree[]=new int[edges.length+2];
    for(int a[]:edges){
    degree[a[0]]++;
    degree[a[1]]++;
    }
        for(int i=1;i<degree.length;i++){
            if(degree[i]==n){
                return i;
            }
        }
        return 0;
    }
}

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        unordered_map<int,int>deg;
        for(auto i:edges){
            deg[i[0]]++;
            deg[i[1]]++;
        }
        
        int size = edges.size();
        
        for(auto it= deg.begin();it!= deg.end(); it++){
            if(it->second == size){
                return it->first;
            }
        }
        return -1;
    }
};